bluetooth: minor enhancements for AES crypto

valeriosetti opened the pull request:
- let `MBEDTLS_AES_ROM_TABLES` be implied and not selected to allow the final user to decide whether they prefer to reduce RAM or ROM footprint
- explicitly select AES-ECB algorithm (i.e. `PSA_WANT_ALG_ECB_NO_PADDING`) for CMAC in BT-crypto

valeriosetti committed:
Bluetooth: make MBEDTLS_AES_ROM_TABLES implied not selected

MBEDTLS_AES_ROM_TABLES let Mbed TLS use precomputed AES tables instead
of generating them at runtime and placing in RAM. It clearly helps
reducing the RAM footprint, but at the same time it increases the ROM
one. In order to let the final user decide which kind of optimization
they prefer, we change the "select" statement with "imply".

Signed-off-by: Valerio Setti <vsetti@baylibre.com>

valeriosetti committed:
Bluetooth: crypto: add select PSA_WANT_ALG_ECB_NO_PADDING

Explicitly enable "PSA_WANT_ALG_ECB_NO_PADDING" to select the AES ECB
mode that it is used in CMAC operation.

Signed-off-by: Valerio Setti <vsetti@baylibre.com>

sjanc commented:
#AutoPTS run zephyr SM

codecoup-tester commented:
AutoPTS Bot results:
<details><summary>Failed tests (1)</summary>SM      SM/PER/PIS/BV-02-C              FAIL</details>
<details><summary>Successful tests (70)</summary>SM      SM/CEN/EKS/BI-01-C              PASS<br>SM      SM/CEN/EKS/BV-01-C              PASS<br>SM      SM/CEN/JW/BI-01-C               PASS<br>SM      SM/CEN/JW/BI-04-C               PASS<br>SM      SM/CEN/JW/BV-05-C               PASS<br>SM      SM/CEN/KDU/BI-04-C              PASS<br>SM      SM/CEN/KDU/BV-04-C              PASS<br>SM      SM/CEN/KDU/BV-05-C              PASS<br>SM      SM/CEN/KDU/BV-06-C              PASS<br>SM      SM/CEN/KDU/BV-10-C              PASS<br>SM      SM/CEN/KDU/BV-11-C              PASS<br>SM      SM/CEN/OOB/BI-01-C              PASS<br>SM      SM/CEN/OOB/BV-01-C              PASS<br>SM      SM/CEN/OOB/BV-03-C              PASS<br>SM      SM/CEN/OOB/BV-09-C              PASS<br>SM      SM/CEN/PIS/BV-02-C              PASS<br>SM      SM/CEN/PIS/BV-03-C              PASS<br>SM      SM/CEN/PKE/BI-01-C              PASS<br>SM      SM/CEN/PKE/BI-02-C              PASS<br>SM      SM/CEN/PKE/BV-01-C              PASS<br>SM      SM/CEN/PKE/BV-04-C              PASS<br>SM      SM/CEN/PROT/BV-01-C             PASS<br>SM      SM/CEN/SCJW/BI-01-C             PASS<br>SM      SM/CEN/SCJW/BV-01-C             PASS<br>SM      SM/CEN/SCJW/BV-04-C             PASS<br>SM      SM/CEN/SCOB/BI-01-C             PASS<br>SM      SM/CEN/SCOB/BI-04-C             PASS<br>SM      SM/CEN/SCOB/BV-01-C             PASS<br>SM      SM/CEN/SCOB/BV-04-C             PASS<br>SM      SM/CEN/SCPK/BI-01-C             PASS<br>SM      SM/CEN/SCPK/BI-02-C             PASS<br>SM      SM/CEN/SCPK/BV-01-C             PASS<br>SM      SM/CEN/SCPK/BV-04-C             PASS<br>SM      SM/CEN/SIGN/BI-01-C             PASS<br>SM      SM/CEN/SIGN/BV-01-C             PASS<br>SM      SM/CEN/SIGN/BV-03-C             PASS<br>SM      SM/PER/EKS/BI-02-C              PASS<br>SM      SM/PER/EKS/BV-02-C              PASS<br>SM      SM/PER/JW/BI-02-C               PASS<br>SM      SM/PER/JW/BI-03-C               PASS<br>SM      SM/PER/JW/BV-02-C               PASS<br>SM      SM/PER/KDU/BI-02-C              PASS<br>SM      SM/PER/KDU/BI-03-C              PASS<br>SM      SM/PER/KDU/BI-04-C              PASS<br>SM      SM/PER/KDU/BV-01-C              PASS<br>SM      SM/PER/KDU/BV-02-C              PASS<br>SM      SM/PER/KDU/BV-03-C              PASS<br>SM      SM/PER/KDU/BV-07-C              PASS<br>SM      SM/PER/KDU/BV-08-C              PASS<br>SM      SM/PER/KDU/BV-09-C              PASS<br>SM      SM/PER/OOB/BI-02-C              PASS<br>SM      SM/PER/OOB/BV-02-C              PASS<br>SM      SM/PER/OOB/BV-04-C              PASS<br>SM      SM/PER/OOB/BV-10-C              PASS<br>SM      SM/PER/PIS/BV-01-C              PASS<br>SM      SM/PER/PKE/BI-03-C              PASS<br>SM      SM/PER/PKE/BV-02-C              PASS<br>SM      SM/PER/PKE/BV-05-C              PASS<br>SM      SM/PER/PROT/BV-02-C             PASS<br>SM      SM/PER/SCJW/BI-02-C             PASS<br>SM      SM/PER/SCJW/BV-02-C             PASS<br>SM      SM/PER/SCJW/BV-03-C             PASS<br>SM      SM/PER/SCOB/BI-02-C             PASS<br>SM      SM/PER/SCOB/BI-03-C             PASS<br>SM      SM/PER/SCOB/BV-02-C             PASS<br>SM      SM/PER/SCOB/BV-03-C             PASS<br>SM      SM/PER/SCPK/BI-03-C             PASS<br>SM      SM/PER/SCPK/BI-04-C             PASS<br>SM      SM/PER/SCPK/BV-02-C             PASS<br>SM      SM/PER/SCPK/BV-03-C             PASS</details>

valeriosetti committed:
Bluetooth: Host: imply MBEDTLS_PSA_P256M_DRIVER_ENABLED rather than select

Allow the user to disable the P256-M driver in case their version of
Mbed TLS does not support this driver.

Signed-off-by: Valerio Setti <vsetti@baylibre.com>

sjanc reviewed (approved):
SM/PER/PIS/BV-02-C is PTS regression

alxelax commented:
@valeriosetti, just got ahead of me for a couple of hours, :) I wanted to create PR with similar changes regarding this macro `MBEDTLS_AES_ROM_TABLES`. `Imply` relation makes using mbedtls easier with custom backends.

frkv reviewed (approved):
LGTM

cvinayak reviewed (approved):
As `BT_CRYPTO` is selected by `BT_LL_SW_SPLIT` and seemly the bsim tests have passed, and the changes LGTM.

cvinayak commented:
Just a nit-pick.... Bluetooth and Host are Title case in Bluetooth Specification... updating the commit message (i.e. no `git diff`) does not require re-approval (just saying) in GitHub PR.

valeriosetti commented:
> Just a nit-pick.... Bluetooth and Host are Title case in Bluetooth Specification... updating the commit message (i.e. no `git diff`) does not require re-approval (just saying) in GitHub PR.

Thanks for the tip. Done ;)
